autoload/maktaba/syscall.vim

```diff
@@ -89,6 +109,58 @@ function! maktaba#syscall#DoCall() abort dict
   return l:return_data
 endfunction
 
+
+""
+" @private
+" @dict Syscall
+" Calls |system()| asynchronously, and invokes a @function(this.callback) once
+" the command completes, passing in stdout, stderr and exit code to it.
+" The specific implementation for @function(#CallAsync).
+function! maktaba#syscall#DoCallAsync() abort dict
+  if empty(v:servername)
+    throw maktaba#error#Message('ShellError', 'Cannot run async commands, no' .
```

Comment:
I feel like for the majority of cases plugin authors would want to fall back to synchronous execution, especially since most console vim users aren't going to be in the habit of always starting vim with `--servername`. If it were `CallAsync({Callback}, {allow_sync_fallback})`, the plugin author could decide what's appropriate for their plugin. Or we could at least have `maktaba#syscall#IsAsyncAvailable()` so they could do something like
```vim
let l:cmd = maktaba#syscall#Create(…)
if maktaba#syscall#IsAsyncAvailable()
  call l:cmd.CallAsync('ResultHandler')
else
  call ResultHandler(l:cmd.Call())
endif
```
Not as convenient as allow_sync_fallback, but also allows custom implementations for when async isn't available. We could even do both if it seems worth it.